File: solutions/go/isogram/1/isogram.go
// Package isogram helps helps to determine certain properties of words
package isogram

import "strings"

// IsIsogram determines if a given word is a nonpattern word
func IsIsogram(word string) bool {
	letters := map[rune]bool{}
	for _, ch := range strings.ToLower(word) {
		if ch == ' ' || ch == '-' {
			continue
		}
		_, was := letters[ch]
		if was {
			return false
		} else {
			letters[ch] = true
		}
	}
	return true
}

File: solutions/ruby/isogram/1/isogram.rb
class Isogram
    def self.isogram? word
        letters = {}
        word.scan(/\w/) do |letter|
            letter = letter.upcase
            letters[letter] ||= 0
            letters[letter] += 1
        end
        letters.all? {|_,num| num<2}
    end
end